pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
-    # TO DO - Implement this function using Ensembl API so that variant location is identified
-    varLoc = '-'
-    return varLoc
+def varOutsideBoundaries(variant):
+    '''Given a variant, determines if variant is outside transcript boundaries'''
+    varGenPos = int(variant["Pos"])
+    varTranscript = variant["Reference_Sequence"]
+    transcriptData = fetch_gene_coordinates(str(varTranscript))
+    varStrand = getVarStrand(variant)
+    if varStrand == "+":
+        txnStart = int(transcriptData["txStart"])
+        txnEnd = int(transcriptData["txEnd"])
+        if varGenPos < txnStart or varGenPos > txnEnd:
+            return True
+    else:
+        txnStart = int(transcriptData["txEnd"])
+        txnEnd = int(transcriptData["txStart"])
+        if varGenPos > txnStart or varGenPos < txnEnd:
+            return True
+
+    return False
+
+
+def varInUTR(variant):
+    '''
+    Given a variant, if variant is inside transcript boundaries, 
+    determines if variant is in 3' or 5' UTR of transcript
+    '''
+    varGenPos = int(variant["Pos"])
+    varTranscript = variant["Reference_Sequence"]
+    transcriptData = fetch_gene_coordinates(varTranscript)
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        if varStrand == "+":
+            tsnStart = int(transcriptData["cdsStart"])
+            tsnEnd = int(transcriptData["cdsEnd"])
+            if varGenPos < tsnStart:
+                return True
+            elif varGenPos > tsnEnd:
+                return True
+        else:
+            tsnStart = int(transcriptData["cdsEnd"])
+            tsnEnd = int(transcriptData["cdsStart"])
+            if varGenPos > tsnStart:
+                return True
+            elif varGenPos < tsnEnd:
+                return True
+        return False
+                
+
+def getExonBoundaries(variant):
+    '''
+    Given a variant, returns the exon boundaries for the variant's transcript in a dictionary with format:
+    key = exon number, value = dictionary with exon start and exon end for specific exon
+    Uses function implemented in calcMaxEntScanMeanStd to get data for variant's transcript
+    '''
+    varTranscript = variant["Reference_Sequence"]
+    transcriptData = fetch_gene_coordinates(str(varTranscript))
+
+    # parse exon starts and exon ends
+    transcriptData["exonStarts"] = re.sub(",(\s)*$", "", transcriptData["exonStarts"])
+    transcriptData["exonEnds"] = re.sub(",(\s)*$", "", transcriptData["exonEnds"])
+    if transcriptData["strand"] == "+":
+        exonStarts = transcriptData["exonStarts"].split(",")
+        exonEnds = transcriptData["exonEnds"].split(",")
+    else:
+        # transcript is on '-' strand
+        exonStarts = list(reversed(transcriptData["exonEnds"].split(",")))
+        exonEnds = list(reversed(transcriptData["exonStarts"].split(",")))
+
+    varExons = {}
+    varExonCount = transcriptData["exonCount"]
+    exonCount = 1
```

Comment:
I added a comment in the code to clarify this, but it starts at 1 because all transcripts start with exon 1, not exon 0 and I'm using the variable exonCount to directly set the exon name as it iterates through the loop. I could theoretically start it at zero, but then there would just be more +1s in the code